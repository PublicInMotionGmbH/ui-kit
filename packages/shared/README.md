# Talixo Shared

Shared data between whole UI Kit.

## How to install

Package is available as `@talixo/shared` in NPM registry, so you can use it in your project
using `npm install @talixo/shared --save` or `yarn add @talixo/shared`.

## Requirements

Your package should additionally have some extra dependencies:

- `classnames: ^2.2.5`

These packages are required by `@talixo/shared`, but you have to install them manually,
to avoid having different versions of these in your application.
