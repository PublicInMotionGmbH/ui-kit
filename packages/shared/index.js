import applyAnyClassNameModifiers from './src/applyAnyClassNameModifiers'
import applyClassNameModifiers from './src/applyClassNameModifiers'
import buildClassName from './src/buildClassName'
import prefix from './src/prefix'

export {
  applyAnyClassNameModifiers,
  applyClassNameModifiers,
  buildClassName,
  prefix
}
