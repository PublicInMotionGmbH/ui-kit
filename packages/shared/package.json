{
  "name": "@talixo/shared",
  "version": "1.0.0",
  "description": "Talixo common configuration",
  "author": "Public in Motion GmbH",
  "main": "index.js",
  "module": "index.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/PublicInMotionGmbH/ui-kit.git"
  },
  "bugs": {
    "url": "https://github.com/PublicInMotionGmbH/ui-kit/issues"
  },
  "devDependencies": {
    "rollup": "^0.57.1"
  },
  "dependencies": {
    "normalize.css": "^8.0.0"
  }
}
