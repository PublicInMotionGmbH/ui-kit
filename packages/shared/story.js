import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'

import createController from './story/createController'

/**
 * Build styles for story
 *
 * @param {object} stylesheet
 * @param {object} options
 * @returns {object}
 */
const styles = (stylesheet, options) => ({
  ...stylesheet,
  infoBody: {
    ...stylesheet.infoBody,
    boxShadow: 'none',
    border: 0,
    padding: 0
  },
  propTableHead: options.propTables && options.propTables.length > 1 ? {

  } : {
    display: 'none'
  },
  source: {
    ...stylesheet.source,
    h1: {
      ...stylesheet.source.h1,
      marginBottom: '15px'
    }
  }
})

/**
 * Create factory for stories
 *
 * @param {string} name
 * @param {object} module
 * @param {object} defaultOptions
 *
 * @returns {function(name: string, description: string, render: function)|function(name: string, description: string, render: function, infoOptions: object)|{ controlled: function(name: string, description: string, render: function, getInitialState: function, infoOptions: object)|function(name: string, description: string, render: function, getInitialState: function, infoOptions: object) }}
 */
export function createStoriesFactory (name, module, defaultOptions = {}) {
  const stories = storiesOf(name, module)

  function addStory (name, description, render, infoOptions = {}) {
    // When rendering function expect any arguments,
    // It means that it's controlled
    if (render.length > 0) {
      render = createController(render)

      infoOptions = {
        ...infoOptions,
        propTablesExclude: [
          render.Controller,
          ...(infoOptions.propTablesExclude || [])
        ]
      }
    }

    // Build options for story
    const options = {
      header: true,
      inline: true,
      text: description,
      ...defaultOptions,
      ...infoOptions
    }

    // Build styles factory
    const stylesFunction = options.styles
      ? stylesheet => options.styles(styles(stylesheet, options), options)
      : stylesheet => styles(stylesheet, options)

    // Create story
    return stories.add(name, withInfo({
      ...options,
      styles: stylesFunction
    })(render))
  }

  return addStory
}

/**
 * Get description from README file
 *
 * @param {string} content
 * @returns {string}
 */
export function getReadmeDescription (content) {
  return content
    .replace(/<p><h1([^>]*)>(.*)<\/h1><\/p>/g, '<h1$1>$2</h1>')
    .replace(/^<h1[^>]*>.*<\/h1>/, '')
    .replace(/(<p>)?<h[1-6][> ][^]*$/i, '')
}
