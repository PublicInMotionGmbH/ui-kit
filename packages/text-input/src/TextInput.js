import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

import calculateInputStyles from '../utils/calculateInputStyles'

// Check if we are currently in environment which supports browser
const isBrowser = typeof document !== 'undefined' && typeof window !== 'undefined'

const moduleName = 'text-input'

/**
 * Component which represents Text Input.
 *
 * @property {object} props
 * @property {string} [props.className]
 * @property {boolean} [props.error]
 * @property {function} [props.onChange]
 * @property {string} [props.placeholder]
 * @property {string} [props.size]
 * @property {object} [props.style]
 *
 * @property {object} state
 * @property {string|null} state.hash
 * @property {object|null} state.suffixStyle
 * @property {object|null} state.inputStyle
 *
 * @property {boolean} hasSuffixInitialized
 *
 * @class
 */
class TextInput extends React.PureComponent {
  constructor (props) {
    super(props)

    this.state = {
      hash: null,
      suffixStyle: { position: 'absolute', visibility: 'hidden', pointerEvents: 'none' },
      inputStyle: null
    }

    this.hasSuffixInitialized = false

    this.onInputChange = this.onInputChange.bind(this)
    this.updateStyles = this.updateStyles.bind(this)
    this.inputRef = this.inputRef.bind(this)
    this.suffixRef = this.suffixRef.bind(this)
  }

  /**
   * Initialize styles (and listeners) for suffix
   */
  initializeSuffixStyles () {
    // Do not work with styles within Node.js environment
    if (!isBrowser) {
      return
    }

    // Do not try initializing suffix styles twice
    if (this.hasSuffixInitialized) {
      return
    }

    // Update suffix and input styles
    this.updateStyles()

    // When component is mounted but styles are still not loaded,
    // It may occur that your suffix is badly positioned.
    window.addEventListener('load', this.updateStyles)
    window.addEventListener('resize', this.updateStyles)

    // Mark initialization as finished
    this.hasSuffixInitialized = true
  }

  /**
   * Tear down styles (and listeners) for suffix
   */
  deinitializeSuffixStyles () {
    // Do not work with styles within Node.js environment
    if (!isBrowser) {
      return
    }

    // Do not try de-initializing suffix styles when they are ready
    if (!this.hasSuffixInitialized) {
      return
    }

    // Remove listener for loaded or resized window
    window.removeEventListener('load', this.updateStyles)
    window.removeEventListener('resize', this.updateStyles)

    // Mark initialization as not done
    this.hasSuffixInitialized = false

    // Update styles to empty them
    this.setState({
      hash: null,
      suffixStyle: { position: 'absolute', visibility: 'hidden', pointerEvents: 'none' },
      inputStyle: null
    })
  }

  /**
   * Position suffix after component is mounted
   */
  componentDidMount () {
    if (this.props.suffix != null) {
      this.initializeSuffixStyles()
    }
  }

  /**
   * Remove listener on unmounting, which is waiting for loaded styles
   */
  componentWillUnmount () {
    this.deinitializeSuffixStyles()
  }

  /**
   * Update suffix when component is updated
   */
  componentDidUpdate () {
    if (this.props.suffix == null) {
      // Try to de-initialize styles when there is no suffix
      this.deinitializeSuffixStyles()
    } else if (this.hasSuffixInitialized) {
      // Update styles if there is already suffix initialized
      this.updateStyles()
    } else {
      // Initialize suffix styles when suffix has been added, but it's not initialized yet
      this.initializeSuffixStyles()
    }
  }

  /**
   * Handle input change
   *
   * @param {SyntheticEvent} e
   */
  onInputChange (e) {
    const { onChange } = this.props

    // Update styles connected to suffix
    this.updateStyles()

    // Trigger change to parent components
    if (onChange) {
      onChange(e.target.value)
    }
  }

  /**
   * Update suffix & input styles
   */
  updateStyles () {
    // Do not try to update styles in Node.js environment
    if (!isBrowser) {
      return
    }

    // Do not update styles while document is not loaded yet
    if (document.readyState !== 'complete' && document.readyState !== 'interactive') {
      return
    }

    // Don't try to calculate styles when there is no suffix or input
    if (!this.input || !this.suffix) {
      return
    }

    // Calculate styles for input & suffix
    const s = calculateInputStyles(this.input, this.suffix)

    // Ignore change if hash of dimensions is the same
    if (s.hash === this.state.hash) {
      return
    }

    // Update component styles
    this.setState({
      hash: s.hash,
      suffixStyle: s.suffix,
      inputStyle: s.input
    })
  }

  /**
   * Cache reference to input element
   *
   * @param {HTMLElement} el
   */
  inputRef (el) {
    this.input = el
  }

  /**
   * Cache reference to suffix element
   *
   * @param {HTMLElement} el
   */
  suffixRef (el) {
    this.suffix = el
  }

  /**
   * Render text input
   *
   * @returns {React.Element}
   */
  render () {
    const { className, error, onChange, size, style, value, suffix, left, right, ...restProps } = this.props
    const { suffixStyle, inputStyle } = this.state

    // Initialize helper variables
    const hasLeft = left != null
    const hasRight = right != null
    const hasSuffix = suffix != null
    const isEmpty = this.input && value == null ? this.input.value === '' : (value == null || value === '')

    // Build classes for `input` element
    const inputClasses = buildClassName(moduleName)

    // Build classes for wrapper
    const wrapperClasses = buildClassName([ moduleName, 'wrapper' ], className, [ size ], {
      'with-left': hasLeft,
      'with-right': hasRight,
      'with-suffix': hasSuffix,
      'empty': isEmpty,
      'error': error
    })

    // Build 'left' element if it's available
    const leftElement = hasLeft ? (
      <span className={buildClassName([ moduleName, 'left' ])}>
        {left}
      </span>
    ) : null

    // Build 'right' element if it's available
    const rightElement = hasRight ? (
      <span className={buildClassName([ moduleName, 'right' ])}>
        {right}
      </span>
    ) : null

    // Build 'suffix' element if it's available, apply styles additionally
    const suffixElement = hasSuffix ? (
      <span
        ref={this.suffixRef}
        className={buildClassName([ moduleName, 'suffix' ])}
        style={suffixStyle}
      >
        {suffix}
      </span>
    ) : null

    // Build text input component
    return (
      <div className={wrapperClasses} style={style}>
        {leftElement}

        <input
          className={inputClasses}
          onChange={this.onInputChange}
          ref={this.inputRef}
          style={inputStyle}
          value={value}
          {...restProps}
        />

        {suffixElement}
        {rightElement}
      </div>
    )
  }
}

TextInput.propTypes = {
  /** Additional class name for text input wrapper */
  className: PropTypes.string,

  /** Indicates that input has error */
  error: PropTypes.bool,

  /** Callback for change event */
  onChange: PropTypes.func,

  /** Size of text input (can be 'small') */
  size: PropTypes.oneOf([ 'small' ]),

  /** Additional input wrapper styling */
  style: PropTypes.object,

  /** Suffix to show after input value */
  suffix: PropTypes.any,

  /** Left side icon or controls */
  left: PropTypes.node,

  /** Right side icon or controls */
  right: PropTypes.node
}

TextInput.defaultProps = {
  error: false,
  type: 'text'
}

export default TextInput
