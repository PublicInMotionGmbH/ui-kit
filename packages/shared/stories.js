import React from 'react'
import { createStoriesFactory } from './story'

// Load first paragraph from README file
const readme = ''

// Create factories for story
const addStory = createStoriesFactory('Typography and basics', module)

// Stories

addStory('Headers', readme, () => (
  <div>
    <h1>h1. Header</h1>
    <h2>h2. Header</h2>
    <h3>h3. Header</h3>
    <h4>h4. Header</h4>
    <h5>h5. Header</h5>
    <h6>h6. Header</h6>

    <h1>h1. Header <small>Secondary text</small></h1>
    <h2>h2. Header <small>Secondary text</small></h2>
    <h3>h3. Header <small>Secondary text</small></h3>
    <h4>h4. Header <small>Secondary text</small></h4>
    <h5>h5. Header <small>Secondary text</small></h5>
    <h6>h6. Header <small>Secondary text</small></h6>
  </div>
))

addStory('Text styling', readme, () => (
  <div>
    <h2>Paragraphs</h2>

    <p>
      Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis.
      Maecenas malesuada elit lectus felis, malesuada ultricies.
      Curabitur et ligula. Ut molestie a, ultricies porta urna.
      Vestibulum commodo volutpat a, convallis ac, laoreet enim. Phasellus fermentum in, dolor.
    </p>

    <p>
      Pellentesque facilisis. Nulla imperdiet sit amet magna.
      Vestibulum dapibus, mauris nec malesuada fames ac turpis velit, rhoncus eu, luctus et interdum adipiscing wisi.
    </p>

    <p>
      Aliquam erat ac ipsum. Integer aliquam purus.
      Quisque lorem tortor fringilla sed, vestibulum id, eleifend justo vel bibendum sapien massa ac turpis
      faucibus orci luctus non, consectetuer lobortis quis, varius in, purus.
      Integer ultrices posuere cubilia Curae, Nulla ipsum dolor lacus, suscipit adipiscing.
      Cum sociis natoque penatibus et ultrices volutpat.
    </p>

    <h2>Inline styling</h2>

    <p>
      We may have some <strong>strong elements</strong> inside our text, as well as some <em>emphasises</em>.
      When we want to calculate 2<sup>3</sup> + x<sub>1</sub>, we may have problems, but it's easy to show it.
      <br />
      In some circumstances, mostly describing our API or other stuff we may also try to show some <code>short code snippet</code>.
      It happens that we may also use <kbd>Ctrl+C</kbd> to hint user with keyboard shortcuts.
    </p>

    <p>
      It may happen that we would like to <mark>highlight</mark> some content inside our dashboards, in case of some search engine.
    </p>

    <p>
      <del>We do not make any mistakes.</del> When we do, we can write it semantically marking that anyway.
    </p>

    <p>
      When we use some shortcuts like <abbr title='Looks good to me'>LGTM</abbr>, we can show such information for user.
    </p>

    <h2>Quotes and testimonials</h2>

    <blockquote>
      This was one of the best experiences I have ever had with a cab company.
      I had problems at the airport and the driver stayed with me for over an hour and helped me sort everything out.
      I would recommend this company to anyone. Thank you for such fabulous service!
    </blockquote>

    <blockquote>
      This was one of the best experiences I have ever had with a cab company.
      I had problems at the airport and the driver stayed with me for over an hour and helped me sort everything out.
      I would recommend this company to anyone. Thank you for such fabulous service!

      <footer>
        <cite>R.M.</cite>
      </footer>
    </blockquote>
  </div>
))

addStory('RTL: Headers', readme, () => (
  <div dir='rtl'>
    <h1>h1. כּוֹתֶרֶת</h1>
    <h2>h2. כּוֹתֶרֶת</h2>
    <h3>h3. כּוֹתֶרֶת</h3>
    <h4>h4. כּוֹתֶרֶת</h4>
    <h5>h5. כּוֹתֶרֶת</h5>
    <h6>h6. כּוֹתֶרֶת</h6>

    <h1>h1. כּוֹתֶרֶת <small>טקסט משני</small></h1>
    <h2>h2. כּוֹתֶרֶת <small>טקסט משני</small></h2>
    <h3>h3. כּוֹתֶרֶת <small>טקסט משני</small></h3>
    <h4>h4. כּוֹתֶרֶת <small>טקסט משני</small></h4>
    <h5>h5. כּוֹתֶרֶת <small>טקסט משני</small></h5>
    <h6>h6. כּוֹתֶרֶת <small>טקסט משני</small></h6>
  </div>
))

addStory('RTL: Text styling', readme, () => (
  <div dir='rtl'>
    <h2>פסקאות</h2>

    <p>
      אחד אל בדפים חרטומים, או פיסול צרפתית ויקיפדיה היא, אם מדריכים הגולשות כתב. ב עוד המחשב ברוכים עקרונות. ספינות מדריכים אקטואליה אחר על.
      זאת אם ביולוגיה אדריכלות, עוד ראשי וקשקש מתמטיקה ב. ויש בה רביעי תרומה לציין. זקוק המדינה גם עוד.
      זאת דת לתרום הבאים נוסחאות, אם פיסיקה סטטיסטיקה אווירונאוטיקה רבה.
    </p>

    <p>
      נפלו למנוע מונחונים או כדי. אחרות תקשורת בה אחר. מיזם יסוד רביעי בה מלא, הבקשה בדפים על זכר. כלל או העזרה ממונרכיה, כלל גם הטבע אנציקלופדיה, אנא אל שמות יסוד.
    </p>

    <p>
תיקונים והגולשים אל בדף, של בדף בישול יוצרים העברית. עזה אם והוא הארץ בידור, תחבורה משופרות עזה אם.
      מה רוסית בארגז הגולשות צ'ט, קבלו מוסיקה ב אחד, מדעי אדריכלות בדף בה. מדע שתפו ויקימדיה אם.
      אדריכלות אווירונאוטיקה אתה של, תרבות ולחבר צ'ט דת.
      לחשבון לויקיפדיה עזה אל, תנך דרכה בכפוף מיוחדים של.
      שתי מה רפואה אודות, כיצד רקטות סטטיסטיקה דת שמו. עוד אקראי וקשקש מיוחדים אם,
      לוח גם העזרה בהתייחסות. מה מונחים אספרנטו קרן. אינו לעריכה של חפש, בה מאמר מוגש אנא.
      בקר הרוח מתמטיקה את, בה החלה יסוד הגולשות קרן. בה לימודים ממונרכיה ואלקטרוניקה עזה,
      חופשית מועמדים מדע של. מדע עזרה אחרות המלצת של, דת יוני אנגלית כתב.
    </p>

    <h2>עיצוב מוטבע</h2>

    <p>
      אולי יש לנו<strong> כמה אלמנטים חזקים בתוך הטקסט </strong>שלנו, כמו גם כמה מדגיש.
      ב במקרים מסוימים, בעיקר לתאר את ה- API שלנו או דברים אחרים אנו עשויים<code> גם לנסות להציג קטע קוד קצר. זה קורה</code>כי אנו יכולים גם >להשתמ
      <kbd>Ctrl+C</kbd> לרמז למשתמש עם קיצורי מקשים.
    </p>

    <p>
      זה יכול לקרות כי אנחנו רוצים להדגיש קצת תוכן בתוך לוחות.<mark> המחוונים שלנו, במקרה של מנוע </mark>החיפוש
    </p>

    <p>
      <del>אנחנו לא עושים שום טעויו</del>ת. כאשר אנו עושים, אנחנו יכולים לכתוב את זה סמנטית סימון כי בכל מקרה.
    </p>

    <p>
      כאשר אנו משתמשים בקיצורי דרך מסוימים כמו <abbr title='נראה לי טוב'>LGTM</abbr>, אנו יכולים להציג מידע כזה עבור המשתמש.
    </p>

    <h2>ציטוטים ועדויות</h2>

    <blockquote>
      זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
    </blockquote>

    <blockquote>
      זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
      <footer>
        <cite>מה-שמו</cite>
      </footer>
    </blockquote>

    <div style={{ display: 'flex' }}>
      <blockquote style={{ marginLeft: '15px' }}>
        זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
  היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
  אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
        <footer>
          <cite>מה-שמו</cite>
        </footer>
      </blockquote>

      <blockquote>
        זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
  היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
  אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
        <footer>
          <cite>מה-שמו</cite>
        </footer>
      </blockquote>
    </div>
  </div>
))
