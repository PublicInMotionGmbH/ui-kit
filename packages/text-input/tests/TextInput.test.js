import React from 'react'
import TextInput from '../src/TextInput'
import { shallow } from 'enzyme'

import { prefix } from '@talixo/shared'

const moduleName = prefix('text-input')

const createProps = (props = {}) => ({
  hasError: false,
  size: '',
  onChange: () => {},
  placeholder: 'Test',
  ...props
})
const defaultProps = createProps()
const createWrapper = (props = defaultProps) => shallow(<TextInput {...props} />)

describe('<TextInput />', () => {
  let wrapper, props

  describe('when rendered', () => {
    beforeAll(() => {
      props = createProps({ className: 'test', hasError: true })
      wrapper = createWrapper(props)
    })

    it('should render properly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when props are passed', () => {
    beforeAll(() => {
      props = createProps({ className: 'test', hasError: true, placeholder: 'test' })
      wrapper = createWrapper(props)
    })

    it('should render placeholder correctly', () => {
      const label = wrapper.find(`.${moduleName}__label`)
      expect(label.text()).toMatch(/^test$/)
    })

    it('should render additional className correctly', () => {
      const input = wrapper.find(`.${moduleName}`)
      expect(input.props().className).toMatch(/(^| )test( |$)/)
    })

    it('should have hasError prop set to true', () => {
      const input = wrapper.find(`.${moduleName}`)
      expect(input.props().className).toMatch(/(^| )talixo-text-input--error( |$)/)
    })
  })

  describe('when user types inside input', () => {
    let input
    const onChange = jest.fn()

    beforeEach(() => {
      wrapper = shallow(<TextInput placeholder='Test' onChange={onChange} />)
      input = wrapper.find(`.${moduleName}`)
      input.simulate('change', { target: { value: 'Test' } })
    })

    it('should call onChange', () => {
      expect(onChange).toHaveBeenCalled()
    })

    it('should update state', () => {
      expect(wrapper.state().inputValue).toEqual('Test')
    })

    it(`should add .${`${moduleName}__label--not-empty`} to label`, () => {
      expect(wrapper.find('label').hasClass(`${moduleName}__label--not-empty`)).toBe(true)
    })

    it(`should remove .${`${moduleName}__label--not-empty`} when input is empty`, () => {
      input.simulate('change', { target: { value: '' } })
      expect(wrapper.find('label').hasClass(`${moduleName}__label--not-empty`)).toBe(false)
    })
  })
})
