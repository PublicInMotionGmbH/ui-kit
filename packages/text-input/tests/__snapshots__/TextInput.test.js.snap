// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render properly 1`] = `
<div
  className="talixo-text-input__wrapper talixo-text-input__wrapper--empty"
>
  <input
    className="talixo-text-input"
    onChange={[Function]}
    style={null}
    type="text"
  />
</div>
`;
